### ISSUE TITLE
Bug : Can't close the script

### ISSUE THREAD

**Issue by gregoiredelzongle** [Hari-Nagarajan/fairgame#21]
Strange behavior appeared today, where CTRL+C or CTRL+D is not able to kill the script anymore (at least on Windows).
Also with the 2060S the script keeps lookings for more to buy after it founds one.

**Comment by johnconnor2020** [comment]
Yup I can recreate that issue, when I first downloaded the script, I was able to kill it with CTRL+C but after updating I cant anymore.

**Comment by Hari-Nagarajan** [comment]
Is this still a problem?

**Comment by yosvany10** [comment]
I was still seeing this behavior with the latest changes tested in powershell and CMD on windows 10.

**Comment by saltynut1** [comment]
Yeah it's not working on my end either. Windows 10 powershell

**Comment by gregoiredelzongle** [comment]
Still happening in #29 , maybe it has something to do with the parallel threading implementation added yesterday

**Comment by zuliwa** [comment]
Still an issue on my end. Maybe the things working in parallel messes with that? How about another thread that just checks for the cancel button?

**Comment by stevenhay** [comment]
It works for me if you keep spamming it, but yeah it doesn't stop immediately.

**Comment by zuliwa** [comment]
This is still an issue for me in the latest version! It's especially annoying when the script tells me I need to rerun it.
I know this is an issue of low priority, but it shouldn't be too hard to fix, right?

**Comment by WillsB3** [comment]
I think this might be because of our use of `ThreadPoolExecutor`. See https://stackoverflow.com/questions/29177490/how-do-you-kill-futures-once-they-have-started

**Comment by gregoiredelzongle** [comment]
You can use CTRL+break to stop the process if CTRL+C doesn't work

**Comment by geekyed** [comment]
Thinking this is to do with sleep() https://www.thetopsites.net/article/51663505.shtml

**Comment by geekyed** [comment]
> I think this might be because of our use of futures and `ThreadPoolExecutor`. See https://stackoverflow.com/questions/29177490/how-do-you-kill-futures-once-they-have-started

tested removing them and just taking the first index of product ids ->  yes this is the cause.

**Comment by DakkJaniels** [comment]
old code
